components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -231,6 +239,58 @@ public boolean doPreSetUserClaimValuesWithID(String userID, Map<String, String>
         return true;
     }
 
+    /**
+     * Validate whether the claim update request is from a provisioned user.
+     *
+     * @param userID           User id.
+     * @param profileName      Profile name.
+     * @param userStoreManager Userstore manager.
+     * @throws UserStoreException if an error occurred while retrieving the user claim list.
+     */
+    private void validateProvisionedUserClaimUpdate(String userID, String profileName,
+                                                    UserStoreManager userStoreManager) throws UserStoreException {
+
+        /*
+        Check whether the config is enabled to edit synced attributes of the provisioned user.
+        If it is enabled, blocking the attribute editing is not required.
+         */
+        if (Boolean.parseBoolean(IdentityUtil.getProperty(SYNCED_ATTRIBUTE_EDITING_ENABLED))) {
+            return;
+        }
+
+        /*
+        Check whether this is an attribute syncing flow by checking the PROVISIONED_USER thread local property.
+        If it is an attribute syncing flow, blocking the attribute editing is not required.
+         */
+        if (IdentityUtil.threadLocalProperties.get().get(FrameworkConstants.JIT_PROVISIONED_USER) != null &&
+                (Boolean) IdentityUtil.threadLocalProperties.get().get(FrameworkConstants.JIT_PROVISIONED_USER)) {
+            return;
+        }
+
+        String username = ((UniqueIDJDBCUserStoreManager) userStoreManager).getUserNameFromUserID(userID);
```

Comment:
why do we need username ? use getUserClaimValuesWithID instead

Reply:
Also don't retrieve all the claims, just pass the PROVISIONED_SOURCE_CLAIM claim and retrieve it.

Reply:
Only doPostGetUserClaimValue method[1] was override in IdentityStoreEventListener. Hense only getUserClaimValues method can be used to get identity claim values. Fixed to only retrieve  PROVISIONED_SOURCE_CLAIM value only with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/367/commits/f4e735453ad2ac354f66a24d63100cecac594701

[1] https://github.com/wso2-extensions/identity-governance/blob/16c7904e76a13b0c317bb8390f3544e958a749fe/components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/listener/IdentityStoreEventListener.java#L213